File: AgentieTursimCS/Domain/Excursie.cs
#nullable enable
using System;

namespace AgentieTurismCS.Domain
{
    public class Excursie : Entity<Guid>
    {
        private string _obiectivTuristic;
        private string _numeFirmaTransport;
        private DateTime _oraPlecare;
        private int _nrLocuriDisponibile;
        private double _pret;

        public Excursie(string obiectivTuristic, string numeFirmaTransport, DateTime oraPlecare, int nrLocuriDisponibile, double pret)
        {
            Id = Guid.NewGuid();
            _obiectivTuristic = obiectivTuristic;
            _numeFirmaTransport = numeFirmaTransport;
            _oraPlecare = oraPlecare;
            _nrLocuriDisponibile = nrLocuriDisponibile;
            _pret = pret;
        }

        public string ObiectivTuristic
        {
            get => _obiectivTuristic;
            set => _obiectivTuristic = value;
        }

        public string NumeFirmaTransport
        {
            get => _numeFirmaTransport;
            set => _numeFirmaTransport = value;
        }

        public DateTime OraPlecare
        {
            get => _oraPlecare;
            set => _oraPlecare = value;
        }

        public int NrLocuriDisponibile
        {
            get => _nrLocuriDisponibile;
            set => _nrLocuriDisponibile = value;
        }

        public double Pret
        {
            get => _pret;
            set => _pret = value;
        }

        public override bool Equals(object? obj)
        {
            if (this == obj) return true;
            if (obj is not Excursie excursie) return false;
            return _nrLocuriDisponibile == excursie._nrLocuriDisponibile && _pret.Equals(excursie._pret) && _obiectivTuristic == excursie._obiectivTuristic && _numeFirmaTransport == excursie._numeFirmaTransport && _oraPlecare == excursie._oraPlecare;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(_obiectivTuristic, _numeFirmaTransport, _oraPlecare, _nrLocuriDisponibile, _pret);
        }

        public override string ToString()
        {
            return $"Excursie{{obiectivTuristic='{_obiectivTuristic}', numeFirmaTransport='{_numeFirmaTransport}', oraPlecare={_oraPlecare}, nrLocuriDisponibile={_nrLocuriDisponibile}, pret={_pret}}}";
        }
    }
}

File: AgentieTurismJava/src/main/java/org/example/agentietursimjava/Domain/Excursie.java
package org.example.agentietursimjava.Domain;

import java.time.LocalDateTime;
import java.util.Objects;
import java.util.UUID;

public class Excursie extends Entity<UUID> {

    private String obiectivTuristic;
    private String numeFirmaTransport;
    private LocalDateTime oraPlecare;
    private int nrLocuriDisponibile;
    private double pret;

    public Excursie(String obiectivTuristic, String numeFirmaTransport, LocalDateTime oraPlecare, int nrLocuriDisponibile, double pret) {
        this.setId(UUID.randomUUID());
        this.obiectivTuristic = obiectivTuristic;
        this.numeFirmaTransport = numeFirmaTransport;
        this.oraPlecare = oraPlecare;
        this.nrLocuriDisponibile = nrLocuriDisponibile;
        this.pret = pret;
    }

    public String getObiectivTuristic() {
        return obiectivTuristic;
    }

    public void setObiectivTuristic(String obiectivTuristic) {
        this.obiectivTuristic = obiectivTuristic;
    }

    public String getNumeFirmaTransport() {
        return numeFirmaTransport;
    }

    public void setNumeFirmaTransport(String numeFirmaTransport) {
        this.numeFirmaTransport = numeFirmaTransport;
    }

    public LocalDateTime getOraPlecare() {
        return oraPlecare;
    }

    public void setOraPlecare(LocalDateTime oraPlecare) {
        this.oraPlecare = oraPlecare;
    }

    public int getNrLocuriDisponibile() {
        return nrLocuriDisponibile;
    }

    public void setNrLocuriDisponibile(int nrLocuriDisponibile) {
        this.nrLocuriDisponibile = nrLocuriDisponibile;
    }

    public double getPret() {
        return pret;
    }

    public void setPret(double pret) {
        this.pret = pret;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Excursie excursie)) return false;
        return nrLocuriDisponibile == excursie.nrLocuriDisponibile && Double.compare(pret, excursie.pret) == 0 && Objects.equals(obiectivTuristic, excursie.obiectivTuristic) && Objects.equals(numeFirmaTransport, excursie.numeFirmaTransport) && Objects.equals(oraPlecare, excursie.oraPlecare);
    }

    @Override
    public int hashCode() {
        return Objects.hash(obiectivTuristic, numeFirmaTransport, oraPlecare, nrLocuriDisponibile, pret);
    }

    @Override
    public String toString() {
        return "Excursie{" +
                "obiectivTuristic='" + obiectivTuristic + '\'' +
                ", numeFirmaTransport='" + numeFirmaTransport + '\'' +
                ", oraPlecare=" + oraPlecare +
                ", nrLocuriDisponibile=" + nrLocuriDisponibile +
                ", pret=" + pret +
                '}';
    }
}

